src/components/Navbar/NavbarLink.tsx

```diff
@@ -44,6 +48,7 @@ export const NavbarLink: FC<NavbarLinkProps> = ({
           theme.disabled[disabled ? 'on' : 'off'],
           className,
         )}
+        onClick={handleClick}
```

Comment:
Ideally I think it would be better to "merge" the user's `onClick` with your custom `handleClick` so that the user's callback does not override the behaviour you're adding

Reply:
@nigellima would it be possible to share the example? as I've seen the other components in `/src/components/*` no where I've seen that the existing code is handling user's `onClick` event as well with our custom `handleClick` function.

Reply:
Something similar to this https://github.com/themesberg/flowbite-react/blob/main/src/components/Dropdown/DropdownItem.tsx#L53

We have an internal callback for the item click but we also make sure that the user's callback is called

Reply:
@nigellima  in that scenario, we've to modify the interface as well to accept the user's `onClick` as well, correct? should we do that? if so then I'll modify and push the code again.

but one question or use-case though, generally do we need an `onClick` event on `Navbar.Link`?

Reply:
it would just be something like this, with the correct types

```javascript
const handleClick = (event) => {
    setIsOpen(!isOpen);
   props.onClick && props.onClick(event)
}
```

Reply:
@nigellima, as we're writing it in TypeScript, so we've to add `onClick` to the interface to accept the props, as it can throw an error during build if we don't pass the `onClick` in props:

https://github.com/themesberg/flowbite-react/blob/main/src/components/Navbar/NavbarLink.tsx#L16


more specifically, in the above interface

Reply:
```tsx
export interface NavbarLinkProps extends ComponentProps<'a'>, Record<string, unknown> {
  active?: boolean;
  as?: ElementType;
  disabled?: boolean;
  href?: string;
  theme?: DeepPartial<FlowbiteNavbarLinkTheme>;
  onClick: () => void
}
```

something this way need to add if we need handle user's `onClick` @nigellima

Reply:
Is it not getting it already from the `ComponentProps<'a'>`?

Reply:
Sorry, it's getting from `ComponentProps<'a'>`.

I've made the changes